tests/cases/conformance/types/typeParameters/typeArgumentLists/instantiationExpressions.ts

```diff
@@ -0,0 +1,171 @@
+}
+
+function f15(f: { new <T>(a: T): T, <U>(a: U, b: number): U[] }) {
+    let fs = f<string>;  // { new (a: string): string; (a: string, b: number): string[]; }
+}
+
+function f16(f: { new <T>(a: T): T, (a: string, b: number): string[] }) {
+    let fs = f<string>;  // new (a: string) => string
+}
+
+function f17(f: { <T>(a: T): T, new (a: string, b: number): string[] }) {
+    let fs = f<string>;  // (a: string) => string
+}
+
+function f20(f: (<T>(a: T) => T) & (<U>(a: U, b: number) => U[])) {
+    let fs = f<string>;  // ((a: string) => string) & ((a: string, b: number) => string[]])
+}
+
+function f21(f: (<T>(a: T) => T) & ((a: string, b: number) => string[])) {
+    let fs = f<string>;  // (a: string) => string
+}
+
+function f22(f: (<T>(a: T) => T) & { x: string }) {
+    let fs = f<string>;  // ((a: string) => string) & { x: string }
+}
+
+function f23(f: { x: string } & { y: string }) {
+    let fs = f<string>;  // Error, no applicable signatures
+}
+
+function f24(f: (new <T>(a: T) => T) & (new <U>(a: U, b: number) => U[])) {
+    let fs = f<string>;  // (new (a: string) => string) & ((a: string, b: number) => string[]])
+}
+
+function f25(f: (new <T>(a: T) => T) & (<U>(a: U, b: number) => U[])) {
+    let fs = f<string>;  // (new (a: string) => string) & ((a: string, b: number) => string[]])
+}
+
+function f26(f: (new <T>(a: T) => T) & ((a: string, b: number) => string[])) {
+    let fs = f<string>;  // new (a: string) => string
+}
+
+function f27(f: (<T>(a: T) => T) & (new (a: string, b: number) => string[])) {
+    let fs = f<string>;  // (a: string) => string
+}
+
+function f30(f: (<T>(a: T) => T) | (<U>(a: U, b: number) => U[])) {
+    let fs = f<string>;  // ((a: string) => string) | ((a: string, b: number) => string[]])
+}
+
+function f31(f: (<T>(a: T) => T) | ((a: string, b: number) => string[])) {
+    let fs = f<string>;  // Error, '(a: string, b: number) => string[]' has no applicable signatures
+}
+
+function f32(f: (<T>(a: T) => T) | { x: string }) {
+    let fs = f<string>;  // ((a: string) => string) | { x: string }
+}
+
+function f33(f: { x: string } | { y: string }) {
+    let fs = f<string>;  // Error, no applicable signatures
+}
+
+function f34(f: (new <T>(a: T) => T) | (new <U>(a: U, b: number) => U[])) {
+    let fs = f<string>;  // (new (a: string) => string) | ((a: string, b: number) => string[]])
+}
+
+function f35(f: (new <T>(a: T) => T) | (<U>(a: U, b: number) => U[])) {
+    let fs = f<string>;  // (new (a: string) => string) | ((a: string, b: number) => string[]])
+}
+
+function f36(f: (new <T>(a: T) => T) | ((a: string, b: number) => string[])) {
+    let fs = f<string>;  // Error, '(a: string, b: number) => string[]' has no applicable signatures
+}
+
+function f37(f: (<T>(a: T) => T) | (new (a: string, b: number) => string[])) {
+    let fs = f<string>;  // Error, 'new (a: string, b: number) => string[]' has no applicable signatures

```

Comment:
Can these all return `fs`, so we can double check how declaration emit serializes these types? And ensure we have some cases where function-level generics are involved? (As signature constraints, parameters, and return types)